### PR TITLE
ansible-doc: remove manual formatting

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Updated version of #78561.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
ansible-doc

### CODE REVIEW

**Review by ssbarnea — APPROVED** [review]
Thanks for taking care of this!

File: lib/ansible/cli/doc.py [hunk 1]
```diff
@@ -1069,9 +1074,10 @@ def add_fields(text, fields, limit, opt_indent, return_values=False, base_indent
 
             text.append("%s%s %s" % (base_indent, opt_leadin, o))
 
+            # description is specifically formated and can either be string or list of strings
             if 'description' not in opt:
                 raise AnsibleError("All (sub-)options and return values must have a 'description' field")
-            if isinstance(opt['description'], list):
+            if isinstance(opt['description'], Sequence) and not isinstance(opt['description'], string_types):
```
Comment (sivel) [inline lib/ansible/cli/doc.py]: We have an `is_sequence` function which you can get from `from ansible.module_utils.common.collections import is_sequence` that handles this distinction.

File: lib/ansible/cli/doc.py [hunk 2]
```diff
@@ -1124,24 +1114,23 @@ def add_fields(text, fields, limit, opt_indent, return_values=False, base_indent
...
                     continue
-                if isinstance(opt[k], string_types):
-                    text.append('%s%s: %s' % (opt_indent, k,
-                                              textwrap.fill(DocCLI.tty_ify(opt[k]),
-                                                            limit - (len(k) + 2),
-                                                            subsequent_indent=opt_indent)))
-                elif isinstance(opt[k], (Sequence)) and all(isinstance(x, string_types) for x in opt[k]):
-                    text.append(DocCLI.tty_ify('%s%s: %s' % (opt_indent, k, ', '.join(opt[k]))))
+
+                if isinstance(opt[k], Sequence) and not isinstance(opt[k], string_types):
```
Comment (sivel) [inline lib/ansible/cli/doc.py]: Same comment as above.  I'd use `is_sequence` here.

File: test/integration/targets/ansible-doc/fakemodule.output [hunk 3]
```diff
@@ -8,7 +8,6 @@ OPTIONS (= is mandatory):
 
 - _notreal
         really not a real option
-        [Default: (null)]
```
Comment (bcoca) [inline test/integration/targets/ansible-doc/fakemodule.output:11]: default should still be shown, even if the formatting changed

File: lib/ansible/cli/doc.py (hunk 1 above)
Comment (felixfontein) [reply lib/ansible/cli/doc.py]: Thanks! Fixed in a1872973319a3a785c663e2e85e9838419882af5.

File: lib/ansible/cli/doc.py (hunk 2 above)
Comment (felixfontein) [reply lib/ansible/cli/doc.py]: Fixed in a1872973319a3a785c663e2e85e9838419882af5.

File: test/integration/targets/ansible-doc/fakemodule.output (hunk 3 above)
Comment (felixfontein) [reply test/integration/targets/ansible-doc/fakemodule.output:11]: Fixed in 8334afd14c07affaa518c86f8ad560375e4cfeb6.

**Comment by bcoca** [comment]
lgtm

**Review by ssbarnea — APPROVED** [review]

**Review by bcoca — CHANGES_REQUESTED** [review]
nvmd

**Review by bcoca — APPROVED** [review]

**Comment by felixfontein** [comment]
@bcoca thanks for your work on this!
@ssbarnea @sivel @bcoca thanks for reviewing and merging!

**Pull request merged**